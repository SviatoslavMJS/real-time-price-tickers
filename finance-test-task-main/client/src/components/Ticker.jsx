import s from "../styles/Ticker.module.css";
import Change from "./Change";
import Company from "./Company";
import Dividend from "./Dividend";
import Exchange from "./Exchage";
import Percent from "./Percent";
import Price from "./Price";
import Yield from "./Yield";

function Ticker(props) {

    const ticker = props.ticker;
    const color = props.color;
    const vector = props.vector;

    return (
        <div className={s.container}>
            <Company ticker={ticker.ticker} color={color} />
            <Exchange exchange={ticker.exchange} />
            <Price price={ticker.price} />
            <Change change={ticker.change} vector={vector} />
            <Percent percent={ticker.change_percent} vector={vector} />
            <Dividend dividend={ticker.dividend} vector={vector} />
            <Yield yield={ticker.yield} vector={vector} />
        </div>
    )
}

export default Ticker;