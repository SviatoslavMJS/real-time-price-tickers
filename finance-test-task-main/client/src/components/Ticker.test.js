import renderer from 'react-test-renderer';
import Ticker from './Ticker';

const startTicker = { "ticker": "AMZN", "exchange": "NASDAQ", "price": 260.34, "change": 128.71, "change_percent": 0.60, "dividend": 0.07, "yield": 0.42, "last_trade_time": "2021-04-30T11:53:21.000Z" };
const changeTicker = { "ticker": "TSLA", "exchange": "NASDAQ", "price": 272.13, "change": 158.76, "change_percent": 0.10, "dividend": 0.96, "yield": 1.00, "last_trade_time": "2021-04-30T11:53:21.000Z" };

test('On change prop vector in Ticker child components must change class names', () => {
    const component = renderer.create(
        <Ticker ticker={startTicker} color={"#666666"} vector={true} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.vector = false;

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.vector = true;

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('On change prop ticker in Ticker child components must change Text content', () => {
    const component = renderer.create(
        <Ticker ticker={startTicker} color={"#666666"} vector={true} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.ticker = changeTicker;

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.ticker = startTicker;

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('On change prop color in Ticker component Company must change style color', () => {
    const component = renderer.create(
        <Ticker ticker={startTicker} color={"#666666"} vector={true} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.color = "#E31937";

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.color = "#666666";

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});