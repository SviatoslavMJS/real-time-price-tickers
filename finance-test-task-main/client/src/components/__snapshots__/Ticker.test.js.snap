// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On change prop color in Ticker component Company must change style color 1`] = `
<div
  className="container"
>
  <div
    className="ticker"
    style={
      Object {
        "backgroundColor": "#666666",
      }
    }
  >
    AMZN
  </div>
  <div
    className="exchange"
  >
    NASDAQ
  </div>
  <div
    className="price"
  >
    $ 260.34
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    128.71
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    0.6 %
  </div>
  <div
    className="green"
  >
    0.07
  </div>
  <div
    className="yieldGreen"
  >
    + 0.42
  </div>
</div>
`;

exports[`On change prop color in Ticker component Company must change style color 2`] = `
<div
  className="container"
>
  <div
    className="ticker"
    style={
      Object {
        "backgroundColor": "#666666",
      }
    }
  >
    AMZN
  </div>
  <div
    className="exchange"
  >
    NASDAQ
  </div>
  <div
    className="price"
  >
    $ 260.34
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    128.71
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    0.6 %
  </div>
  <div
    className="green"
  >
    0.07
  </div>
  <div
    className="yieldGreen"
  >
    + 0.42
  </div>
</div>
`;

exports[`On change prop color in Ticker component Company must change style color 3`] = `
<div
  className="container"
>
  <div
    className="ticker"
    style={
      Object {
        "backgroundColor": "#666666",
      }
    }
  >
    AMZN
  </div>
  <div
    className="exchange"
  >
    NASDAQ
  </div>
  <div
    className="price"
  >
    $ 260.34
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    128.71
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    0.6 %
  </div>
  <div
    className="green"
  >
    0.07
  </div>
  <div
    className="yieldGreen"
  >
    + 0.42
  </div>
</div>
`;

exports[`On change prop ticker in Ticker child components must change Text content 1`] = `
<div
  className="container"
>
  <div
    className="ticker"
    style={
      Object {
        "backgroundColor": "#666666",
      }
    }
  >
    AMZN
  </div>
  <div
    className="exchange"
  >
    NASDAQ
  </div>
  <div
    className="price"
  >
    $ 260.34
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    128.71
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    0.6 %
  </div>
  <div
    className="green"
  >
    0.07
  </div>
  <div
    className="yieldGreen"
  >
    + 0.42
  </div>
</div>
`;

exports[`On change prop ticker in Ticker child components must change Text content 2`] = `
<div
  className="container"
>
  <div
    className="ticker"
    style={
      Object {
        "backgroundColor": "#666666",
      }
    }
  >
    AMZN
  </div>
  <div
    className="exchange"
  >
    NASDAQ
  </div>
  <div
    className="price"
  >
    $ 260.34
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    128.71
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    0.6 %
  </div>
  <div
    className="green"
  >
    0.07
  </div>
  <div
    className="yieldGreen"
  >
    + 0.42
  </div>
</div>
`;

exports[`On change prop ticker in Ticker child components must change Text content 3`] = `
<div
  className="container"
>
  <div
    className="ticker"
    style={
      Object {
        "backgroundColor": "#666666",
      }
    }
  >
    AMZN
  </div>
  <div
    className="exchange"
  >
    NASDAQ
  </div>
  <div
    className="price"
  >
    $ 260.34
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    128.71
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    0.6 %
  </div>
  <div
    className="green"
  >
    0.07
  </div>
  <div
    className="yieldGreen"
  >
    + 0.42
  </div>
</div>
`;

exports[`On change prop vector in Ticker child components must change class names 1`] = `
<div
  className="container"
>
  <div
    className="ticker"
    style={
      Object {
        "backgroundColor": "#666666",
      }
    }
  >
    AMZN
  </div>
  <div
    className="exchange"
  >
    NASDAQ
  </div>
  <div
    className="price"
  >
    $ 260.34
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    128.71
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    0.6 %
  </div>
  <div
    className="green"
  >
    0.07
  </div>
  <div
    className="yieldGreen"
  >
    + 0.42
  </div>
</div>
`;

exports[`On change prop vector in Ticker child components must change class names 2`] = `
<div
  className="container"
>
  <div
    className="ticker"
    style={
      Object {
        "backgroundColor": "#666666",
      }
    }
  >
    AMZN
  </div>
  <div
    className="exchange"
  >
    NASDAQ
  </div>
  <div
    className="price"
  >
    $ 260.34
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    128.71
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    0.6 %
  </div>
  <div
    className="green"
  >
    0.07
  </div>
  <div
    className="yieldGreen"
  >
    + 0.42
  </div>
</div>
`;

exports[`On change prop vector in Ticker child components must change class names 3`] = `
<div
  className="container"
>
  <div
    className="ticker"
    style={
      Object {
        "backgroundColor": "#666666",
      }
    }
  >
    AMZN
  </div>
  <div
    className="exchange"
  >
    NASDAQ
  </div>
  <div
    className="price"
  >
    $ 260.34
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    128.71
  </div>
  <div
    className="green"
  >
    <svg
      className="svg_up"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    0.6 %
  </div>
  <div
    className="green"
  >
    0.07
  </div>
  <div
    className="yieldGreen"
  >
    + 0.42
  </div>
</div>
`;
